cpp/src/io/utilities/datasource.hpp

```diff
@@ -87,6 +97,54 @@ class datasource {
    * @return bool True if there is data, False otherwise
    **/
   virtual bool empty() const { return size() == 0; }
+
+  /**
+   * @brief Returns whether the data source returns fixed-address buffers
+   *
+   * @return bool True if get_buffer addresses only depend on offset, False otherwise
```

Comment:
What is the get_buffer that is mentioned here (and used below)? Doesn't seem to be a member of this class? Do you mean get_device_buffer?

Reply:
10 lines above at line 84. datasource::get_buffer() is the main function to get bytes out of the datasource.

Reply:
browser find doesn't work when GitHub collapses the unchanged lines. :)